core/rec-ARM/rec_arm.cpp

```diff
@@ -2114,37 +2114,50 @@ void ngen_Compile(RuntimeBlockInfo* block,bool force_checks, bool reset, bool st
 	reg.OpBegin(&block->oplist[0],0);
 
 	//scheduler
-	if (force_checks)
+	if (force_checks && settings.dynarec.SmcCheckLevel != NoCheck)
```

Comment:
Can we pass this as an option (maybe extend force checks) instead of reaching out to the config from the backend?